src/main/java/frc/robot/subsystems/SwerveDriveSubsystem.java

```diff
@@ -102,17 +104,30 @@ public void resetOdometry(Pose2d pose) {
    *                      field. Positive is counterclockwise.
    */
   public void drive(double xSpeed, double ySpeed, double rot, boolean fieldRelative) {
+    double rotation = rot;
+
+    // corrects the heading of the robot to prevent it from drifting
+    if (rotation != 0) { // if rotating
+      m_headingCorrectionPID.setSetpoint(Utils.normalizeAngle(m_gyro.getRotation2d().getRadians(), 2 * Math.PI));
+      SmartDashboard.putString("Heading Correction", "setting setpoint");
+    } else if (xSpeed != 0 || ySpeed != 0) { // else if translating
+      rotation = m_headingCorrectionPID.calculate(Utils.normalizeAngle(m_gyro.getRotation2d().getRadians(), 2 * Math.PI));
+      SmartDashboard.putString("Heading Correction", "correcting heading");
+    } else {
+      SmartDashboard.putString("Heading Correction", "not running");
```

Comment:
```suggestion
      SmartDashboard.putString("Heading Correction", "setting setpoint");
      m_headingCorrectionPID.setSetpoint(Utils.normalizeAngle(m_gyro.getRotation2d().getRadians(), 2 * Math.PI));
```
Continuing to set the setpoint while stationary should fix the problem

Reply:
But what if you're translating and rotating at the same time, and then you stop rotating but you keep translating?

Reply:
it wouldn't fix that but I think it would fix the problem where you rotate and then translate